test/unit/auth/credential.spec.ts

```diff
@@ -519,6 +569,36 @@ describe('Credential', () => {
       expect(isApplicationDefault(c)).to.be.true;
     });
 
+    describe("FIREBASE_EMULATOR_CREDENTIALS", () => {
+
+      let googleCreds: any = undefined;
+
+      before(() => {
+        googleCreds = process.env.GOOGLE_APPLICATION_CREDENTIALS;
+        delete process.env.GOOGLE_APPLICATION_CREDENTIALS;
```

Comment:
I think just unsetting this variable within the test case should be enough. We used to do this trick of restoring the env variables in the past, but that's not really needed for our tests.

Reply:
My concern is that if a test fails before we get to the line where it unsets, it can cause downstream failures.

Reply:
Unset at the beginning of the test case, which is what `before()` would do anyway. Just don't worry about restoring the old value.